### PR TITLE
Feature/name removal

### PR DESCRIPTION
**Pull request by albrja**
### Removes names from simulation and replaces them with name id columns.
- *Category*: 
- *JIRA issue*: [MIC-3658](https://jira.ihme.washington.edu/browse/MIC-3658)
- *Research reference*: N/A

### Changes and notes
-Removes name columns from state table and replaces them with name_id columns for first, middle, and last name.
-Updates column names in observers.

### Verification and Testing
Ran simulation for 250,000 and saw improvement in performance.

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/components/population.py [hunk 1]
```diff
@@ -126,10 +114,11 @@ def generate_initial_population(self, pop_data: SimulantData) -> None:
         # drop non-unique household_id
         pop = pop.drop(columns="census_household_id")
 
-        # give names
-        first_and_middle = self.name_generator.generate_first_and_middle_names(pop)
-        last_names = self.name_generator.generate_last_names(pop)
-        pop = pd.concat([pop, first_and_middle, last_names], axis=1)
+        # give name ids
+        pop["first_name_id"] = pop.index
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/components/population.py:118]: So what is the plan here? At what point do these become actual id numbers?

Comment (albrja) [reply src/vivarium_census_prl_synth_pop/components/population.py:118]: This is just going to be the id number.  We will handle in post-processing what these actually map to.  This takes away all generation and sampling for names out of the simulation.  So different ids may map to the same name.  We just want to make sure that if we know simulants will share a name that they will have the same name_id.  That is what the next ticket will implement for last name matching.

File: src/vivarium_census_prl_synth_pop/components/population.py [hunk 2]
```diff
@@ -331,8 +320,8 @@ def initialize_newborns(self, pop_data: SimulantData) -> None:
         new_births["born_in_us"] = True
 
         # add first and middle names
-        names = self.name_generator.generate_first_and_middle_names(new_births)
-        new_births = pd.concat([new_births, names], axis=1)
+        new_births["first_name_id"] = new_births.index
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/population.py:323]: Does this value increment from 0 or from the last index value in the pop table? If the latter, this looks fine

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_census_prl_synth_pop/components/population.py (hunk 2 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/components/population.py:321]: Last index in the pop table.

**Pull request merged**